pkgs/development/python-modules/qiskit-aer/default.nix

```
@@ -3,65 +3,87 @@
 , buildPythonPackage
 , fetchFromGitHub
 , cmake
+, cvxpy
 , cython
 , numpy
 , openblas
 , pybind11
 , scikit-build
-, cvxpy
+, spdlog
   # Check Inputs
 , qiskit-terra
 , pytestCheckHook
+, python
 }:
 
 buildPythonPackage rec {
   pname = "qiskit-aer";
-  version = "0.3.4";
+  version = "0.4.1";
 
   disabled = pythonOlder "3.5";
 
   src = fetchFromGitHub {
     owner = "Qiskit";
     repo = pname;
     rev = version;
-    sha256 = "0d9s8j06s2dsxq127m21b309pvqb8fj9zia70jc0clsarrawg314";
+    fetchSubmodules = true; # fetch muparserx and other required libraries
+    sha256 = "1j2pv6jx5dlzanjp1qnf32s53d8jrlpv96nvymznkcnjvqn60gv9";
   };
 
   nativeBuildInputs = [
     cmake
-    cython
     scikit-build
   ];
 
   buildInputs = [
     openblas
+    spdlog
   ];
 
   propagatedBuildInputs = [
     numpy
-    pybind11
     cvxpy
+    cython  # generates some cython files at runtime that need to be cython-ized
+    pybind11
   ];
 
   prePatch = ''
     # remove dependency on PyPi cmake package, which isn't in Nixpkgs
-    substituteInPlace setup.py --replace ", 'cmake'" ""
+    substituteInPlace setup.py --replace "'cmake'" ""
+  '';
 
-    # Add tests & test config to distribution
-    echo "recursive-include test *" >> MANIFEST.in
-    echo "include *.conf" >> MANIFEST.in
+  # Move out of build dir to source dir to run setup.py
+  postConfigure = ''
+    cd ..
   '';
 
-  dontUseCmakeConfigure = true; # tries to run setup.py via cmake, which doesn't work here
   cmakeFlags = [
     "-DBUILD_TESTS=True"
+    "-DAER_THRUST_BACKEND=OMP"
   ];
 
-  pythonImportsCheck = [ "qiskit.providers.aer" "qiskit.providers.aer.backends.qasm_simulator" ];
-  checkInputs = [ qiskit-terra pytestCheckHook ];
-  dontUseSetuptoolsCheck = true;
+  # Needed to find qiskit.providers.aer modules in cython. This exists in GitHub, don't know why it isn't copied by default
+  postFixup = ''
+    touch $out/${python.sitePackages}/qiskit/__init__.pxd
+  '';
+
+  # *** Testing ***
+
+  pythonImportsCheck = [ "qiskit.providers.aer" "qiskit.providers.aer.backends.qasm_simulator" "qiskit.providers.aer.backends.controller_wrappers" ];
```

this is getting a little long
```suggestion
  pythonImportsCheck = [ 
    "qiskit.providers.aer" 
    "qiskit.providers.aer.backends.qasm_simulator" 
    "qiskit.providers.aer.backends.controller_wrappers"
  ];
```